rust/kernel/platdev.rs

```diff
@@ -49,13 +54,22 @@ impl Registration {
             return Err(Error::EINVAL);
         }
         this.pdrv.driver.name = name.as_ptr() as *const c_types::c_char;
+        if let Some(tbl) = of_match_table {
+            let ptr = tbl.into_pointer();
+            this.of_table = Some(ptr);
+            this.pdrv.driver.of_match_table = ptr.cast();
+        }
```

Comment:
Maybe ``this.of_table`` should be a ``Option<OfMatchTable>``. I don't understand the need of a raw pointer here. This could fix the ``drop`` later on ?

Reply:
It's because structures that are shared with kernel structs using raw pointers, use the `PointerWrapper` abstraction/trait. This trait ensures that the struct is consumed when turned into a kernel pointer. After the call to `into_pointer()`, there is no `tbl` object left to store anywhere !